base/stat.jl

```diff
@@ -301,7 +301,7 @@ function filemode_string(mode)
         end
         complete && write(str, "-")
     end
-    return String(take!(str))
+    return unsafe_takestring!(str)
```

Comment:
Same as above: this is safe because we are only calling methods we own, `write(::IOBuffer, ::Char)` and `write(::IOBuffer, ::String)` and don't return `str`.